src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,369 @@
+    will list files within provided local path with -f option and will prepend
+    each file with a \"Y\" if file exceeds threshold value and with \"N\" if
+    it does not.
+
+ 2. Compress files specified in provided local path with -f option that exceed
+    1M in size. This script will check if current local system is a VM host, if
+    so, compression will not be executed.
+
+ 3. Archive files specified in provided local path with -f option to provided
+    remote path destination.
+
+ 4. Create and display cart_logtest log files created by running cart_logtest.py
+    script. Output of running the script will be redirected to *_ctestlog.log
+    files.
+
+ Options:
+      -z        Compress files
+      -t        Threshold value to determine classification of big logs
+      -a        Remote path to scp files to i.e. server-A:/path/to/file
+      -f        Local path to files for archiving/compressing/size checking
+      -c        Enable running cart_logtest.py
+      -x        Enable performing compression when running on VM system
+      -v        Display commands being executed
+      -h        Display this help and exit
+"
+}
+
+#######################################
+# CONSTANT variables.
+#######################################
+# shellcheck disable=SC2046,SC2086
+CART_LOGTEST_PATH="$(dirname $(readlink -f ${0}))/cart/cart_logtest.py"
+
+#######################################
+# Check if the files exist on this host.
+# Arguments:
+#   $1: local logs to process
+# Returns:
+#   0: Files exist or no matches were found
+#   1: Missing the log files argument
+#######################################
+check_files_input() {
+    if [ -n "${1}" ]; then
+        # shellcheck disable=SC2086
+        files=$(ls -d ${1} 2> /dev/null)
+        files_rc=$?
+        if [ -n "${files}" ] && [ "${files_rc}" -eq 0 ]; then
+            echo "Files found that match ${1}."
+        else
+            echo "No files matched ${1}. Nothing to do."
+            exit 0
+        fi
+    else
+        echo "Please specify -f option."
+        exit 1
+    fi
+}
+
+#######################################
+# Convert human readable size values to respective byte value.
+# Arguments:
+#   size: size value to convert to bytes. i.e. 5mb, 40gb, 6k, 10g
+# Returns:
+#   converted byte value.
+#######################################
+human_to_bytes() {
+    h_size=${1,,}
+
+    declare -A units=(
+        [b]=1
+        [kb]=$(python -c "print (2**10)")
+        [k]=$(python -c "print (2**10)")
+        [mb]=$(python -c "print (2**20)")
+        [m]=$(python -c "print (2**20)")
+        [gb]=$(python -c "print (2**30)")
+        [g]=$(python -c "print (2**30)")
+    )
+    if [[ "${h_size,,}" =~ ([0-9.]+)([kbmgt]+) ]]; then
+        val="${BASH_REMATCH[1]}"
+        unit="${BASH_REMATCH[2]}"
+        (( val = val*${units[${unit}]} ))
+        echo "${val}"
+    else
+        echo "Invalid value provided: ${1}"
+        return 1
+    fi
+}
```

Comment:
This is a re-implementation of [`numfmt`](https://www.gnu.org/software/coreutils/manual/html_node/numfmt-invocation.html).  Better to use tools on hand if they serve purpose as it reduces the defect and maintenance footprint.

Reply:
Ah, I wasn't aware of numfmt. I'll modify the code to use it.